sdk-workflows/src/main/java/io/dapr/workflows/runtime/WorkflowRuntimeBuilder.java

```diff
@@ -18,11 +18,18 @@
 import io.dapr.workflows.Workflow;
 import io.dapr.workflows.internal.ApiTokenClientInterceptor;
 import io.grpc.ClientInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.HashSet;
+import java.util.Set;
 
 public class WorkflowRuntimeBuilder {
   private static volatile WorkflowRuntime instance;
   private DurableTaskGrpcWorkerBuilder builder;
   private static ClientInterceptor WORKFLOW_INTERCEPTOR = new ApiTokenClientInterceptor();
+  private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowRuntimeBuilder.class);
+  private volatile Set<String> activitySet = new HashSet<String>(); 
```

Comment:
Make it final and use synchronizedSet instead: https://www.baeldung.com/java-synchronized-collections. Volatile on a Set does not guarantee thread-safety of operations in the collection.

Reply:
gentle ping - @addjuarez looks like this is almost complete, one last update and please merge in master 😃